fbw-a380x/src/systems/instruments/src/EWD/shared/ArincValueProvider.ts

```diff
@@ -3,27 +3,87 @@
 // SPDX-License-Identifier: GPL-3.0
 
 import { EventBus } from '@microsoft/msfs-sdk';
-import { Arinc429Register } from '@flybywiresim/fbw-sdk';
+import { Arinc429Register, Arinc429Word } from '@flybywiresim/fbw-sdk';
 import { EwdSimvars } from './EwdSimvarPublisher';
 
 export interface Arinc429Values {
   sat: Arinc429Register;
   slatsFlapsStatus: Arinc429Register;
   slatsPosition: Arinc429Register;
   flapsPosition: Arinc429Register;
+  cpiomB1AgsDiscrete: Arinc429Word;
+  cpiomB2AgsDiscrete: Arinc429Word;
+  cpiomB3AgsDiscrete: Arinc429Word;
+  cpiomB4AgsDiscrete: Arinc429Word;
+  cpiomBAgsDiscrete: Arinc429Word;
 }
 export class ArincValueProvider {
   private sat = Arinc429Register.empty();
 
+  private cpiomB1AgsDiscrete = new Arinc429Word(0);
+
+  private cpiomB2AgsDiscrete = new Arinc429Word(0);
+
+  private cpiomB3AgsDiscrete = new Arinc429Word(0);
+
+  private cpiomB4AgsDiscrete = new Arinc429Word(0);
+
+  private cpiomBToUse = 1;
+
   constructor(private readonly bus: EventBus) {}
 
   public init() {
     const publisher = this.bus.getPublisher<Arinc429Values>();
     const subscriber = this.bus.getSubscriber<EwdSimvars>();
+    this.cpiomBToUse = this.determineCpiomBToUse();
```

Comment:
Is it intended that this only looks once at instrument load?

Reply:
No it was not 😄 I've changed it now to avoid this (hopefully).